xds/src/main/java/io/grpc/xds/XdsNameResolverProvider.java

```diff
@@ -43,8 +43,11 @@
 
   private static final String SCHEME = "xds";
 
-  @Override
-  public XdsNameResolver newNameResolver(URI targetUri, Args args) {
+  /**
+   * Allows injecting bootstrapOverride to the name resolver.
+   * */
+  public XdsNameResolver newNameResolver(URI targetUri, Args args,
```

Comment:
Who would call this method? I think we want `bootstrapOverride` to be passed to a constructor/factoryMethod for the provider itself.

Reply:
Right, i didn't test it. 
The `bootstrapOverride` needs to be saved here and then passed to `xdsNameResolver` when channel creates a new resolver.